konnectivity-client/pkg/client/client.go

```diff
@@ -116,14 +117,15 @@ func (t *grpcTunnel) serve(c clientConn) {
 			if !ok {
 				klog.V(1).Infoln("DialResp not recognized; dropped")
 			} else {
-				ch <- dialResult{
-					err:    resp.Error,
-					connid: resp.ConnectID,
-				}
 				// We should only get DialResp once. If we get the DialResp
-				// twice, we will get an exception trying to write to a closed
-				// channel.
-				close(ch)
+				// multiple times, the subsequent responses will be ignored
+				t.handleDialOnce.Do(func() {
```

Comment:
We want the code to fail to expose if we ever got two DIAL_RSP.

Can you check if a goroutine panics, does that interrupt the execution of the main code?

If not, then we don't need to do anything.

If it does, then we can call recover() in server() to handle the exception.

Reply:
Updated. The recover function is a no-op if no panic is hit, so it shouldn't interrupt execution.